### PR TITLE
fix(vue): Move `authStatus` logic to composable

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes
This PR moves authStatus and hub listener logic from the UI to AuthServices so that it can be used without Authenticator UI.

#### Description of how you validated changes

New unit, e2e tests, and manual validation:

https://github.com/aws-amplify/amplify-ui/assets/43682783/c09ed5b7-f894-40f6-adc0-61ffa20d63e1

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .changeset/smart-cats-mix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/ui-vue': patch
+---
+
+fix(angular): separate `authStatus` logic from UI
```
Comment (wlee221) [inline .changeset/smart-cats-mix.md:5]:
```suggestion
fix(vue): separate `authStatus` logic from UI
```

**Review by ioanabrooks — APPROVED** [review]

**Review by zchenwei — APPROVED** [review]
LGTM

File: packages/vue/jest.config.cjs [hunk 2]
```diff
@@ -10,9 +10,9 @@ module.exports = {
   coverageThreshold: {
     global: {
       branches: 22,
-      functions: 20,
-      lines: 45,
-      statements: 44,
+      functions: 22,
+      lines: 46,
```
Comment (reesscot) [inline packages/vue/jest.config.cjs:14]: nice!

Comment (wlee221) [reply packages/vue/jest.config.cjs:14]: Slowly going back up... 😛

**Review by ioanabrooks — APPROVED** [review]

File: packages/e2e/features/ui/components/authenticator/auth-status.feature [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
+the authenticated status of the application.
+
+This feature tests that `authStatus` is available and updated even if Authenticator
+UI is not rendered in the DOM tree.
+
   Background:
     Given I'm running the example "/ui/components/authenticator/auth-status"
 
-  @angular @react
+  @angular @react @vue
```
Comment (ioanabrooks) [inline packages/e2e/features/ui/components/authenticator/auth-status.feature:12]: nice!

File: packages/vue/src/composables/__tests__/useAuthenticator.spec.ts [hunk 4]
```diff
@@ -59,7 +60,7 @@ describe('useAuthenticator', () => {
       state: mockState.value,
     });
 
-    expect(wrapper.vm.authStatus).toBe('authenticated');
```
Comment (reesscot) [inline packages/vue/src/composables/__tests__/useAuthenticator.spec.ts:62]: Why did this test previously assume you were authenticated?

**Review by reesscot — APPROVED** [review]

File: packages/vue/src/composables/__tests__/useAuthenticator.spec.ts (hunk 4 above)
Comment (wlee221) [reply packages/vue/src/composables/__tests__/useAuthenticator.spec.ts:62]: Oh, if you look at line 34 above i changed the mock value to `unauthenticated`. That probably was misleading... :D

**Review by zchenwei — APPROVED** [review]

**Pull request merged**